slither/tools/read_storage/read_storage.py

```diff
@@ -234,22 +364,143 @@ def get_slot_values(self, slot_info: SlotInfo) -> None:
         )
         logger.info(f"\nValue: {slot_info.value}\n")
 
-    def get_all_storage_variables(self, func: Callable = None) -> None:
+    def get_all_storage_variables(self, func: Callable = lambda x: x) -> None:
         """Fetches all storage variables from a list of contracts.
         kwargs:
             func (Callable, optional): A criteria to filter functions e.g. name.
         """
         for contract in self.contracts:
-            self._target_variables.extend(
-                filter(
-                    func,
-                    [
-                        (contract, var)
-                        for var in contract.state_variables_ordered
-                        if not var.is_constant and not var.is_immutable
-                    ],
-                )
+            for var in contract.state_variables_ordered:
+                if func(var):
+                    if not var.is_constant and not var.is_immutable:
+                        self._target_variables.append((contract, var))
+                    elif var.is_constant and var.type == ElementaryType("bytes32"):
```

Comment:
Should this also be guarded by the unstructured flag?

Reply:
It could be. I just figured it didn't hurt to add the variables to `self._constant_storage_slots` anyway, since this part is not too computationally complex.